bin/matrix

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+set -euo pipefail
+
+# Version utility for comparing versions from
+# https://stackoverflow.com/questions/4023830/how-to-compare-two-strings-in-dot-separated-version-format-in-bash
+function version { echo "$@" | awk -F. '{ printf("%d%03d%03d%03d\n", $1,$2,$3,$4); }'; }
+
+# Store current Ruby version
+original_ruby_version=$(cat .ruby-version)
+
+{
+  # Determine employed Ruby versions
+  readarray -t versions < <(cat .github/workflows/test.yml | grep ruby: | awk '{$1=$1};1' | sed 's/- ruby: //' | sort | uniq)
+  echo "Detected Ruby versions:"
+  for version in "${versions[@]}"
+  do
+    echo "- $version"
+  done
+  echo ""
+
+  for version in "${versions[@]}"
+  do
+    rbenv local "$version"
+
+    ruby_version=$(ruby -v)
+    rubygems_version=$(gem -v)
+    bundler_version=$(bundler -v | sed 's/Bundler version //')
+
+    echo "====================="
+    echo "Target Ruby version: $version"
+    echo ""
+    echo "Ruby: $ruby_version"
+    echo "rubygems: $rubygems_version"
+    echo "Bundler: $bundler_version"
+    echo "====================="
+    echo ""
+
+    if [ $(version $rubygems_version) -lt $(version "3.3.0") ]; then
+      echo "Please ensure that your rubygems version is > 3.3.0 for Ruby $ruby_version!"
+      echo "Install newest version:"
+      echo "gem update --system"
+      echo "Install specific version:"
+      echo "gem update --system <version>"
+      exit 1
```

Comment:
why exit if you install the desired version anyway?

Reply:
I don't install the versions. That would require me to determine the correct version. I only echo a helpful message to the terminal so that users of the script don't have to search for the commands.